src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -13,53 +13,64 @@ const LOG_AREA = 'PlainWebSocketsTransport';
 
 const NAME = 'plainWebSockets';
 
-// Internal webSocket readyState possible values.
-// eslint-disable-next-line no-unused-vars
-const STATE_INTERNAL_CONNECTING = 0;
-const STATE_INTERNAL_OPEN = 1;
-const STATE_INTERNAL_CLOSING = 2;
-const STATE_INTERNAL_CLOSED = 3;
+const socketCloseCodes = {
+    NORMAL_CLOSURE: 1000,
+    TOKEN_EXPIRED: 1002,
+};
+
+const CLOSE_REASON_DESTROY = 'Normal Close due to connection destroy action';
 
 const DEFAULT_RECONNECT_DELAY = 2000;
 const DEFAULT_RECONNECT_LIMIT = 10;
 
 const NOOP = () => {};
 
-// Normal closure; the connection successfully completed whatever purpose for which it was created.
-// Ref. https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent#Status_codes
-const CLOSE_STATUS_CODE = 1000;
-const CLOSE_REASON_DESTROY = 'Normal Close due to connection destroy action';
-
 // -- Local methods section --
 
 function normalizeWebSocketUrl(url) {
     return url.replace('http://', 'ws://').replace('https://', 'wss://');
 }
 
-function getWebSocketUrl() {
-    return normalizeWebSocketUrl.call(
-        this,
-        `${this.connectionUrl}${this.query}`,
-    );
+function connect() {
+    try {
+        const url = normalizeWebSocketUrl.call(
+            this,
+            `${this.connectionUrl}${this.query}`,
+        );
+
+        log.debug(LOG_AREA, 'Creating WebSocket connection', { url });
+        const socket = new WebSocket(url);
+
+        socket.binaryType = 'arraybuffer';
+        socket.onopen = handleSocketOpen.bind(this);
+        socket.onmessage = handleSocketMessage.bind(this);
+        socket.onclose = handleSocketClose.bind(this);
+
+        this.socket = socket;
+    } catch (error) {
+        handleFailure.call(this, {
+            message: 'Failed to setup webSocket connection',
+        });
+    }
 }
 
-function destroySocket(socket) {
+function disconnect() {
+    const socket = this.socket;
     if (!socket) {
         return;
     }
+
     socket.onopen = null;
-    socket.onerror = null;
     socket.onmessage = null;
     socket.onclose = null;
-    socket.close(CLOSE_STATUS_CODE, CLOSE_REASON_DESTROY);
+    socket.close(socketCloseCodes.NORMAL_CLOSURE, CLOSE_REASON_DESTROY);
+
+    this.socket = null;
 }
 
 function reconnect() {
     if (this.reconnectCount >= DEFAULT_RECONNECT_LIMIT) {
-        handleFailure.call(this, {
-            message:
-                'Transport reached maximum amount of reconnects allowed till last successful connection.',
-        });
+        this.stop();

```

Comment:
This will enable infinite retry for plain websockets
should this be controlled by flag ?
whether to fallback to signalR or retry again

Reply:
@lukeapage Can you please check this comment ?